### PR TITLE
Closes #2250 - Refactor TerminateTaskAccTest to use test-api

### LINKED ISSUE #2250: Refactor TerminateTaskAccTest to use test-api

### PR DESCRIPTION
**Pull request by jamesrdi**
https://sonarcloud.io/summary/new_code?id=jamesrdi_taskana&branch=TSK-2250

---
Release Notes:

```

```
 
### For the submitter:
- [ ] I updated the [documentation](https://taskana.atlassian.net/wiki/spaces/TAS/overview) and will supply links to the specific files
- [x] I did not update the [documentation](https://taskana.atlassian.net/wiki/spaces/TAS/overview)
- [x] I included a link to the [SonarCloud branch analysis](https://taskana.atlassian.net/wiki/spaces/TAS/pages/1019969636/SonarCloud+Integration)
- [ ] After integration of the PR, I added a description of changes on the [current release notes](https://taskana.atlassian.net/wiki/spaces/TAS/pages/1281392672/Current+Release+Notes+Taskana)
- [x] I did not update the [current release notes](https://taskana.atlassian.net/wiki/spaces/TAS/pages/1281392672/Current+Release+Notes+Taskana)
- [ ] I put the ticket in review
- [ ] After integration of the pull request, I verified our [bluemix test environment](http://taskana.mybluemix.net/taskana) is not broken

### Verified by the reviewer:
- [ ] Commit message format → TSK-XXX: Your commit message.
- [ ] Submitter's update to [documentation](https://taskana.atlassian.net/wiki/spaces/TAS/overview) is sufficient
- [ ] SonarCloud analysis meets our standards
- [ ] Update of the [current release notes](https://taskana.atlassian.net/wiki/spaces/TAS/pages/1281392672/Current+Release+Notes+Taskana) reflects changes
- [ ] PR fulfills the ticket
- [ ] Edge cases and unwanted side effects are tested
- [ ] Readability

### CODE REVIEW

File: lib/taskana-core-test/src/test/java/acceptance/task/complete/TerminateTaskAccTest.java [hunk 1]
```diff
@@ -0,0 +1,179 @@
...
+@TaskanaIntegrationTest
+class TerminateTaskAccTest {
+  @TaskanaInject TaskService taskService;
+  @TaskanaInject ClassificationService classificationService;
+  @TaskanaInject WorkbasketService workbasketService;
+  @TaskanaInject CurrentUserContext currentUserContext;
+
+  ClassificationSummary defaultClassificationSummary;
+  WorkbasketSummary defaultWorkbasketSummary;
+  ObjectReference defaultObjectReference;
```
Comment (ryzheboka) [inline lib/taskana-core-test/src/test/java/acceptance/task/complete/TerminateTaskAccTest.java:42]: Why do you define it here without usind? Can you delete defaultObjectReference?

File: lib/taskana-core-test/src/test/java/acceptance/task/complete/TerminateTaskAccTest.java [hunk 2]
```diff
@@ -0,0 +1,179 @@
...
+    ThrowingCallable call = () -> taskService.terminateTask(task.getId());
+
+    NotAuthorizedException e = catchThrowableOfType(call, NotAuthorizedException.class);
+    assertThat(e.getCurrentUserId()).isEqualTo(currentUserContext.getUserid());
+    assertThat(e.getRoles()).containsExactlyInAnyOrder(TaskanaRole.ADMIN, TaskanaRole.TASK_ADMIN);
+  }
+
+  @WithAccessId(user = "taskadmin")
+  @Test
+  void should_ThrowException_When_TerminateTaskWithTaskStateTerminated() throws Exception {
```
Comment (ryzheboka) [inline lib/taskana-core-test/src/test/java/acceptance/task/complete/TerminateTaskAccTest.java:140]: Please create a TestFactory for the three end states

File: lib/taskana-core-test/src/test/java/acceptance/task/complete/TerminateTaskAccTest.java [hunk 3]
```diff
@@ -0,0 +1,163 @@
...
+        TaskBuilder.newTask()
+            .classificationSummary(defaultClassificationSummary)
+            .workbasketSummary(defaultWorkbasketSummary)
+            .state(TaskState.CLAIMED)
+            .primaryObjRef(DefaultTestEntities.defaultTestObjectReference().build())
+            .buildAndStore(taskService);
+
+    taskService.terminateTask(task.getId());
+
+    Task readTask = taskService.getTask(task.getId());
```
Comment (ryzheboka) [inline lib/taskana-core-test/src/test/java/acceptance/task/complete/TerminateTaskAccTest.java:94]: You don't need to read the Task again. Just get the return value from terminateTask

**Review by ryzheboka — APPROVED** [review]
